### PR TITLE
Add window size conditions for paywall component overrides

### PR DESCRIPTION
**Pull request by joshdholtz**
### Checklist
- [x] If applicable, unit tests
- [x] If applicable, create follow-up issues for `purchases-ios` and hybrids (iOS: RevenueCat/purchases-ios#7645; hybrids need no changes)

### Motivation
Let one paywall adapt its layout to the window it renders in (tablets, foldables, multi-window) — the Android side of upcoming responsive paywall rules in the dashboard.

### Description
- New override conditions: `WindowWidthRule`, `WindowHeightRule`, `WindowAspectRatioRule` (`window_width_condition` etc.) — operators `>=` `>` `<=` `<` `=`, values in dp, ratio is width / height (>1 = landscape)
- Conditions AND together (`width >= 700` + `height >= 480` = large windows, not landscape phones)
- Evaluated against the paywall's own measured bounds (same as iOS): a paywall in a sheet, dialog, or pane matches its own size. An unbounded axis (fit-content sheet height) falls back to the window dimension. Re-evaluates live on rotation/resize/multi-window
- Older SDKs decode them as `Unsupported` → default layout; unknown window size never matches
- Snapshot coverage twice over: Compose previews (CI snapshot pipeline) + Paparazzi goldens — phone portrait stacked, phone landscape stacked (height floor), tablets side by side

⚠️ Reviewer notes:
- Selection gap (a rule hiding the currently selected package leaves it selected and purchasable) is fixed in the stacked follow-up #4198, mirroring iOS's reconcile-on-resize.
- Earlier revisions evaluated the app window; after review feedback this now measures the paywall's own bounds, matching iOS.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Medium Risk**
> Wide changes to override resolution and layout roots across paywall rendering, though behavior is gated on new dashboard rules and covered by tests; package selection still ignores window-size rules (follow-up noted for hidden-selection edge cases).
> 
> **Overview**
> Adds **responsive paywall overrides** driven by measured paywall size (not the full app window), aligned with iOS: width, height, and aspect-ratio rules with `>=` / `>` / `<=` / `<` / `=` on dp values, AND-combined within an override.
> 
> The **purchases** layer gains `ComparisonOperator` and three new serialized condition types (`window_width_condition`, etc.), marked as rules so bad payloads still fall back safely. **RevenueCat UI** measures bounds via `MeasurePaywallBounds` into `paywallBoundsDp`, threads that through `ConditionContext` into every component state, and evaluates comparisons (including epsilon for `=`) while size is unknown → no match until the first measure.
> 
> Standard and **workflow** paywalls wrap their scaffolds in the measurer so headers see bounds on frame one. Previews, Paparazzi snapshots, and broad unit tests cover deserialization, operator edge cases, and a split-layout fixture (stacked vs side-by-side at 700×480dp).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit a25cc1a60030a6af0f6ed4be36792412cb833c45. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/ScreenCondition.kt [hunk 1]
```diff
@@ -1,8 +1,25 @@
...
+ * [currentWindowSize]'s window metrics, so it updates automatically on rotation,
+ * resize, and multi-window changes — the same source the window size classes
+ * derive from.
+ */
+@JvmSynthetic
+@Composable
+internal fun currentWindowDpSize(): DpSize = with(LocalDensity.current) {
+    currentWindowSize().toSize().toDpSize()
+}
+
```
Comment (JZDesign) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/ScreenCondition.kt:23]: doesn't this measure the activity window, not the paywall bounds?

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/PresentedPartial.kt [hunk 2]
```diff
@@ -202,7 +240,7 @@ private fun ComponentOverride.Condition.Variable.matchesValue(
     else -> false
 }
 
-private const val STATE_NUMBER_COMPARISON_EPSILON = 1e-10
+private const val NUMBER_COMPARISON_EPSILON = 1e-10
```
Comment (JZDesign) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/PresentedPartial.kt:250]: Sol has issues with 1e-10 It says equality is effectively unusable because of floating point precision. Since this is pre-existing, it's not for this PR, but it is good to note.

**Review by JZDesign — COMMENTED** [review]
I think this PR has the same issue as the iOS one, where a selected package that gets hidden is still selected and purchaseable

**Comment by joshdholtz** [comment]
> I think this PR has the same issue as the iOS one, where a selected package that gets hidden is still selected and purchaseable

@JZDesign You right you right! it's the (known gap called out in the PR description) 

Rather than throwing it into this PR, I'll create a follow up stacking it on top of this one that mirrors iOS's `reconcileSelection` so... stay tuned 😊

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/ScreenCondition.kt (hunk 1 above)
Comment (joshdholtz) [reply ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/ScreenCondition.kt:23]: Fixed!

**Review by JZDesign — DISMISSED** [review]
~~I unresolved a comment because I accidentally posted a reply in the wrong spot. But over all I think this makes sense with those follow ups in place~~ this was supposed to be the iOS PR comment I got turned around

**Review by JZDesign — APPROVED** [review]
I think this looks good. I do think we need to address the 1e-10 equality thing before we ship this, but that can go in a different PR

**Comment by joshdholtz** [comment]
> I think this looks good. I do think we need to address the 1e-10 equality thing before we ship this, but that can go in a different PR

@JZDesign Just made changes for this!
